akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java

```diff
@@ -0,0 +1,72 @@
+package jdocs.akka.typed.testing.sync;
+
+//#imports
+import akka.actor.typed.*;
+import akka.actor.typed.javadsl.*;
+import akka.typed.testkit.*;
+//#imports
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+public class BasicSyncTestingTest extends JUnitSuite {
+
+  //#child
+  public static Behavior<String> childActor = Actor.immutable((ctx, msg) -> Actor.same());
+  //#child
+
+  //#under-test
+  interface Command { }
+  public static class CreateAChild implements Command { }
+  public static class CreateAnAnonymousChild implements Command { }
+  public static class SayHello implements Command {
+    private final ActorRef<String> who;
+
+    public SayHello(ActorRef<String> who) {
+      this.who = who;
+    }
+  }
+
+  public static Behavior<Command> myBehaviour = Actor.immutable(Command.class)
+    .onMessage(CreateAChild.class, (ctx, msg) -> {
+      ctx.spawn(childActor, "child");
+      return Actor.same();
+    })
+    .onMessage(CreateAnAnonymousChild.class, (ctx, msg) -> {
+      ctx.spawnAnonymous(childActor);
+      return Actor.same();
+    })
+    .onMessage(SayHello.class, (ctx, msg) -> {
+      msg.who.tell("hello");
+      return Actor.same();
+    }).build();
+  //#under-test
+
+
+  @Test
+  public void testSpawning() {
+    //#test-child
+    BehaviorTestkit<Command> test = BehaviorTestkit.apply(myBehaviour);
+    test.run(new CreateAChild());
+    test.expectEffect(new Effect.Spawned(childActor, "child", Props.empty()));
```

Comment:
As discussed elsewhere we might want to return the `effects` instead and then use ordinary testing framework facilities for assertions. Probably convenient with both styles, e.g. this would be nice:
```
test.expectSpawned(actorName = "child")
```

Reply:
We do have both an API for returning one effect or all the effects so users can do this. I just didn't add examples yet.

Regarding the per effect assertions, i did start adding them but there are so many effects it becomes a big API, I think that is okay, WDYT?